handler/graphql.go

```diff
@@ -271,35 +298,58 @@ func GraphQL(exec graphql.ExecutableSchema, options ...Option) http.HandlerFunc
 			}
 		}
 		if doc == nil {
-			var qErr gqlerror.List
-			doc, qErr = gqlparser.LoadQuery(exec.Schema(), reqParams.Query)
-			if len(qErr) > 0 {
-				sendError(w, http.StatusUnprocessableEntity, qErr...)
+			var gqlErr *gqlerror.Error
+			var listErr gqlerror.List
+
+			ctx = cfg.tracer.StartOperationParsing(ctx)
```

Comment:
its probably time to break this into a separate function so the ends can be deferred. This currently looks like a recipe for bugs soup.